### PR TITLE
replacing axios to native request to handle proxy issue

### PR DESCRIPTION
**Pull request by roshan04**

### CODE REVIEW

File: bin/helpers/buildArtifacts.js [hunk 1]
```diff
@@ -95,33 +97,36 @@ const downloadAndUnzip = async (filePath, fileName, url) => {
...
-        writer.close();
-        reject(err);
-      });
-      writer.on('close', async () => {
-        if (!error) {
-          await unzipFile(filePath, fileName);
-          fs.unlinkSync(tmpFilePath);
-          resolve(true);
+  return new Promise(async (resolve, reject) => {
+    try {
```
Comment (nagpalkaran95) [inline bin/helpers/buildArtifacts.js:101]: why do we need `try` here?

File: bin/helpers/buildArtifacts.js [hunk 2]
```diff
@@ -135,86 +140,126 @@ const unzipFile = async (filePath, fileName) => {
...
-    reporter = args.reporter;
-  } else if(cypressJSON !== undefined){
-    reporter = cypressJSON.reporter;
-  }
+    let reporter = null;
+    if(!utils.isUndefined(args.reporter)) {
+      reporter = args.reporter;
+    } else if(cypressJSON !== undefined){
+      reporter = cypressJSON.reporter;
+    }
```
Comment (nagpalkaran95) [inline bin/helpers/buildArtifacts.js:153]: please keep indentation same (2 spaces as everyplace else)

File: bin/helpers/buildArtifacts.js [hunk 3]
```diff
@@ -135,86 +140,126 @@ const unzipFile = async (filePath, fileName) => {
...
+    let errorCode = null;
+
+    try {
+      let buildDetails = null;
+      request.get(options, async function (err, resp, body) {
+        if(err) {
+          utils.sendUsageReport(bsConfig, args, err, Constants.messageTypes.ERROR, 'api_failed_build_artifacts', null, rawArgs);
+          process.exitCode = Constants.ERROR_EXIT_CODE;
+        } else {
+          try {
```
Comment (nagpalkaran95) [inline bin/helpers/buildArtifacts.js:221]: try inside try? not a good approach. outside try should take care of failures

**Review by nagpalkaran95 — CHANGES_REQUESTED** [review]

**Comment by nagpalkaran95** [comment]
let's update reporting methods too

File: bin/helpers/utils.js [hunk 4]
```diff
@@ -17,7 +17,6 @@ const usageReporting = require("./usageReporting"),
   config = require("../helpers/config");
 
 const request = require('request');
-const axios = require("axios");
```
Comment (nagpalkaran95) [inline bin/helpers/utils.js:20]: We need to remove the lib from package.json as well

**Review by nagpalkaran95 — CHANGES_REQUESTED** [review]

File: bin/helpers/buildArtifacts.js [hunk 5]
```diff
@@ -156,65 +157,95 @@ const sendUpdatesToBstack = async (bsConfig, buildId, args, options, rawArgs) =>
...
         process.exitCode = Constants.ERROR_EXIT_CODE;
       } else {
-        messageType = Constants.messageTypes.SUCCESS;
-        message = Constants.userMessages.DOWNLOAD_BUILD_ARTIFACTS_SUCCESS.replace('<build-id>', buildId).replace('<user-path>', process.cwd());
-        logger.info(message);
+        try {
+          buildDetails = JSON.parse(body);
+          if(resp.statusCode != 200) {
+            logger.error('Downloading the build artifacts failed.');
+            logger.error(`Error: Request failed with status code ${resp.statusCode}`)
```
[the remark this answers is not in the document]
Comment (roshan04) [reply bin/helpers/buildArtifacts.js:214]: No had synced with product, we will be just sending the stacktrace to BQ, and for user we will be showing the statuscode.

**Review by nagpalkaran95 — APPROVED** [review]

**Review by nagpalkaran95 — APPROVED** [review]

**Comment by pranavj1001** [comment]
Closing this PR on account of https://github.com/browserstack/browserstack-cypress-cli/pull/196

**Pull request closed without merging**